lazy_theta_star_planner/src/lazy_theta_star2.cpp

```diff
@@ -0,0 +1,257 @@
+//  Copyright 2020 Anshumaan Singh
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http:// www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "lazy_theta_star_planner/lazy_theta_star2.h"
+#include <vector>
+
+namespace lazyThetaStar
+{
+
+LazyThetaStar::LazyThetaStar()
+{
+  sizeX = 0;
+  sizeY = 0;
+}
+
+bool LazyThetaStar::getPath(std::vector<coords<world_pts>> & raw_path)
+{
+  RCLCPP_INFO(node_->get_logger(), "Path Planning Begins....");
+
+  initializeStuff();
+
+  std::cout << src.x << '\t' << src.y << '\t' << dst.x << '\t' << dst.y << '\n';
+
+  data.push_back({src.x, src.y, 0, dist(src.x, src.y, dst.x, dst.y),
+      0, 1, dist(src.x, src.y, dst.x, dst.y)});
+  addIndex(src.x, src.y, id_gen);
+  pq.push({id_gen, &(data[id_gen].f)});
+
+  id_gen++;
+  id curr_id = 0;
+
+  if (!losCheck(src.x, src.y, dst.x, dst.y)) {
```

Comment:
Maybe try unnesting this. Have `if (losCheck()) {... straight path stuff ... return};` then do the non-LOS stuff so that you don't have this entire while loop nested in this condition. Making things as flat as possible is preferable